### PR TITLE
Fix trie bugs

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
- Fixed a bug that wrongfully marked for eviction a node when reduceNode() was called for a branch node.
- The cache for oldHashes and newHashes are now reset every time after a Commit() occurs.

### CODE REVIEW

File: data/trie/patriciaMerkleTrie.go [hunk 1]
```diff
@@ -130,6 +132,10 @@ func (tr *patriciaMerkleTrie) Update(key, value []byte) error {
 		}
 		tr.root = newRoot
 		tr.oldHashes = append(tr.oldHashes, oldHashes...)
+
+		for i := range oldHashes {
```
Comment (sasurobert) [inline data/trie/patriciaMerkleTrie.go:136]: for _, hash := range oldHashes

File: data/trie/patriciaMerkleTrie.go [hunk 2]
```diff
@@ -145,6 +151,10 @@ func (tr *patriciaMerkleTrie) Update(key, value []byte) error {
...
```
Comment (sasurobert) [inline data/trie/patriciaMerkleTrie.go:155]: for _, hash := range oldHashes

File: data/trie/patriciaMerkleTrie.go [hunk 3]
```diff
@@ -369,6 +378,11 @@ func (tr *patriciaMerkleTrie) ResetOldHashes() [][]byte {
 	oldHashes := tr.oldHashes
 	tr.oldHashes = make([][]byte, 0)
 	tr.oldRoot = make([]byte, 0)
+
+	for i := range oldHashes {
```
Comment (sasurobert) [inline data/trie/patriciaMerkleTrie.go:382]: for _, hash := range oldHashes

File: data/trie/patriciaMerkleTrie.go [hunk 4]
```diff
@@ -394,6 +408,10 @@ func (tr *patriciaMerkleTrie) GetDirtyHashes() (data.ModifiedHashes, error) {
 		return nil, err
 	}
 
+	for hash := range dirtyHashes {
```
Comment (sasurobert) [inline data/trie/patriciaMerkleTrie.go:411]: for _, hash := range oldHashes

File: data/trie/patriciaMerkleTrie.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply data/trie/patriciaMerkleTrie.go:136]: Done.

File: data/trie/patriciaMerkleTrie.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply data/trie/patriciaMerkleTrie.go:155]: Done.

File: data/trie/patriciaMerkleTrie.go (hunk 3 above)
Comment (BeniaminDrasovean) [reply data/trie/patriciaMerkleTrie.go:382]: Done.

File: data/trie/patriciaMerkleTrie.go (hunk 4 above)
Comment (BeniaminDrasovean) [reply data/trie/patriciaMerkleTrie.go:411]: DirtyHashes is a map[string]struct{}, so it is ok like it is.

**Pull request merged**